assets/src/data/eventespresso/filter-state/index.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * External dependencies
+ */
+import { Map } from 'immutable';
+import createSelector from 'rememo';
+import { registerStore } from '@wordpress/data';
+
+const DEFAULT_STATE = {};
+const SET_FILTER = 'SET_FILTER';
+
+const getState = ( state, context ) => {
+	return typeof state[ context ] === 'undefined' ?
+		Map() :
+		state[ context ];
+};
+
+const filterStateReducer = ( state = DEFAULT_STATE, action ) => {
+	const { type, context, filter, value } = action;
+	if ( type === SET_FILTER ) {
+		return {
+			...state,
+			[ context ]: getState( state, context ).set( filter, value ),
+		};
+	}
+	return state;
+};
+
+const setFilter = ( context, filter, value ) => {
+	return {
+		type: SET_FILTER,
+		context,
+		filter,
+		value,
+	};
+};
+
+const getFilter = createSelector(
+	( state, context, filter, defaultValue ) => {
+		return getState( state, context ).get( filter, defaultValue );
+	},
+	( state, context, filter, defaultValue ) => [
+		getState( state, context ).get( filter, defaultValue ),
```

Comment:
Hmm, I don't think `defaultValue` should be passed through on this argument as this argument is used by rememo to determine whether there is a state change and if there is to "recalculate" by calling the original selector (which is the first argument on `createSelector`).  And actually (again, I know you were going by my original example), if we're only dealing with scalar values here we don't need to use rememo at all.  So you could simply write this selector as:

```js
const getFilter = ( state, context, filter, defaultValue ) => {
	return getState( state, context ).get( filter, defaultValue );
};
```